pkg/operator/ceph/csi/template/rbd/csi-rbdplugin-provisioner-dep.yaml

```diff
@@ -268,10 +298,56 @@ spec:
               mountPath: /csi
           imagePullPolicy: {{ .ImagePullPolicy }}
         {{ end }}
+        {{ if .CSILogRotation }}
+        - args:
+          - |
+            echo "Starting the csi-logrotate-sidecar"
+            mkdir -p /logrotate-config;
+            echo '/var/lib/cephcsi/log/*.log {
+                daily
+                missingok
+                rotate 7
+                compress
+                copytruncate
+                notifempty
+            }' > /logrotate-config/csi;
+            echo "File creation container completed";
+            LOG_ROTATE_CEPH_FILE=/logrotate-config/csi
+            PERIODICITY={{ .CSILogRotationPeriod }}
+            if [ "$PERIODICITY" != "" ]; then
+              echo "replace default daily with given user input"
+              sed --in-place "s/daily/$PERIODICITY/g" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            LOG_MAX_SIZE={{ .CSILogRotationMaxSize }}
+            if [ "$LOG_MAX_SIZE" != "0" ]; then
+              sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            while true; do
+              logrotate --verbose "$LOG_ROTATE_CEPH_FILE"
+              sleep 15m
+            done
+          command:
+          - /bin/sh
+          - -c
+          image: {{ .CSIPluginImage }}
+          imagePullPolicy: IfNotPresent
+          name: log-collector
+          volumeMounts:
+          - mountPath:  /logrotate-config
+            name: csi-logs-logrotate
+          - mountPath: /var/lib/cephcsi/log/
```

Comment:
why /var/lig is mounted here not /var/log?

Reply:
so we decided to keep the log files in this folder `/var/lib/cephcsi/log/`

if you want us to change I can.